File: challenges/medium/06_counter_closure/counter.py
def make_counter(start=0):
    """
    Returns a dict of counter operations sharing the same internal state.

    Supported ops (to be extended):
      "inc": increment count by N, return new count
      "dec": decrement count by N, return new count  ← TODO: implement
    """
    count = start

    def increment(by=1):
        count += by  # something is off here
        return count

    def decrement(by=1):
        pass  # TODO: subtract by from count and return count

    return {"inc": increment, "dec": decrement}

File: challenges/medium/12_counter_closure_js/counter.js
function makeCounter(start = 0) {
  const count = start;    // bug: const — cannot be reassigned inside the closure

  function increment(by = 1) {
    count += by;           // TypeError: Assignment to constant variable
    return count;
  }

  function decrement(by = 1) {
    // TODO: subtract by from count and return count
  }

  return { inc: increment, dec: decrement };
}

module.exports = { makeCounter };
